procedures/katello/branch.md.erb

```diff
@@ -3,18 +3,26 @@
 ## Release Owner
 
 - [ ] Announce upcoming branching to discourse development category a month before
-- [ ] Start extracting i18n strings and pushing the changes into master git branches so Transifex can pick it up
 - [ ] Update Katello Transifex translations:
   - [ ] Create a Transifex account and join the Foreman team
   - [ ] Spin up a Foreman and Katello installation
-  - [ ] Install Transifex client `pip3 install transifex-client --user`
-  - [ ] [Configure the Transifex client](https://projects.theforeman.org/projects/foreman/wiki/Translating#How-to-pull-translations)
+  - [ ] Install Transifex client: `pip3 install transifex-client --user` (make sure to pass `--user` flag!)
+  - [ ] Configure `~/.transifexrc` [as outlined here](https://projects.theforeman.org/projects/foreman/wiki/Translating#How-to-pull-translations)
```

Comment:
The linked doc tells you to `yum install` rather than `pip3 install` `transifex-client`, which caused me all sorts of problems.  so I wanted to make this a bit more clear.

Reply:
Since there is a client in both Fedora and Debian, I think that would be preferred over `pip3`. Those typically do survive distro updates while `pip3` often doesn't.

Reply:
Can you elaborate why `yum install` caused problems?

Reply:
I don't remember the specific error, but I have to do this again anyway, so I'll try to reproduce and get back with more details..

Reply:
> Can you elaborate why `yum install` caused problems?

Just got a chance to run this again.  After installing with `yum -y install transifex-client gettext make intltool`, here is the result of `make -C locale tx-update`:

```
[vagrant@centos7-katello-devel-stable katello]$ make -C locale tx-update
make: Entering directory `/home/vagrant/katello/locale'
tx pull -f
Traceback (most recent call last):
  File "/usr/bin/tx", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 3007, in <module>
    working_set.require(__requires__)
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 728, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 626, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: python-slugify
make: *** [tx-pull] Error 1
make: Leaving directory `/home/vagrant/katello/locale'
```

I then tried pip3 install, but I didn't think to yum remove first:
```
[vagrant@centos7-katello-devel-stable katello]$ pip3 install transifex-client --user
Collecting transifex-client
  Downloading https://files.pythonhosted.org/packages/1c/25/1824710f70aa6665b97b5faa84fd4fe16da20e72ecf1598e39457b2a651f/transifex-client-0.14.2.tar.gz (211kB)
    100% |████████████████████████████████| 215kB 3.3MB/s 
Requirement already satisfied: urllib3<2.0.0,>=1.24.2 in /usr/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: six<2.0.0 in /usr/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: requests<3.0.0,>=2.19.1 in /usr/lib/python3.6/site-packages (from transifex-client)
Collecting python-slugify<5.0.0 (from transifex-client)
  Downloading https://files.pythonhosted.org/packages/9f/42/e336f96a8b6007428df772d0d159b8eee9b2f1811593a4931150660402c0/python-slugify-4.0.1.tar.gz
Collecting gitpython<4.0.0 (from transifex-client)
  Downloading https://files.pythonhosted.org/packages/d7/6d/0528adaff6229c5cd85feb84366e1cf3130d86c69d0acea02fe12b5d79c4/GitPython-3.1.15-py3-none-any.whl (163kB)
    100% |████████████████████████████████| 174kB 8.6MB/s 
Requirement already satisfied: chardet<5,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Requirement already satisfied: idna<3,>=2.5 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Collecting text-unidecode>=1.3 (from python-slugify<5.0.0->transifex-client)
  Downloading https://files.pythonhosted.org/packages/a6/a5/c0b6468d3824fe3fde30dbb5e1f687b291608f9473681bbf7dabbf5a87d7/text_unidecode-1.3-py2.py3-none-any.whl (78kB)
    100% |████████████████████████████████| 81kB 3.6MB/s 
Requirement already satisfied: typing-extensions>=3.7.4.0 in /usr/lib/python3.6/site-packages (from gitpython<4.0.0->transifex-client)
Collecting gitdb<5,>=4.0.1 (from gitpython<4.0.0->transifex-client)
  Downloading https://files.pythonhosted.org/packages/ea/e8/f414d1a4f0bbc668ed441f74f44c116d9816833a48bf81d22b697090dba8/gitdb-4.0.7-py3-none-any.whl (63kB)
    100% |████████████████████████████████| 71kB 7.4MB/s 
Collecting smmap<5,>=3.0.1 (from gitdb<5,>=4.0.1->gitpython<4.0.0->transifex-client)
  Downloading https://files.pythonhosted.org/packages/68/ee/d540eb5e5996eb81c26ceffac6ee49041d473bc5125f2aa995cf51ec1cf1/smmap-4.0.0-py2.py3-none-any.whl
Installing collected packages: text-unidecode, python-slugify, smmap, gitdb, gitpython, transifex-client
  Running setup.py install for python-slugify ... done
  Running setup.py install for transifex-client ... done
```

It seems like it found the python-slugify that was missing.  But I got the same error.

I then yum removed transifex-client and pip3 installed again:

```
[vagrant@centos7-katello-devel-stable katello]$ sudo yum remove transifex-client
Loaded plugins: fastestmirror
Resolving Dependencies
--> Running transaction check
---> Package transifex-client.noarch 0:0.13.4-1.el7 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

============================================================================================================================================
 Package                                Arch                         Version                              Repository                   Size
============================================================================================================================================
Removing:
 transifex-client                       noarch                       0.13.4-1.el7                         @epel                       562 k

Transaction Summary
============================================================================================================================================
Remove  1 Package

Installed size: 562 k
Is this ok [y/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Erasing    : transifex-client-0.13.4-1.el7.noarch                                                                                     1/1 
  Verifying  : transifex-client-0.13.4-1.el7.noarch                                                                                     1/1 

Removed:
  transifex-client.noarch 0:0.13.4-1.el7                                                                                                    

Complete!
[vagrant@centos7-katello-devel-stable katello]$ pip3 install transifex-client --user
Requirement already satisfied: transifex-client in /home/vagrant/.local/lib/python3.6/site-packages
Requirement already satisfied: urllib3<2.0.0,>=1.24.2 in /usr/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: six<2.0.0 in /usr/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: requests<3.0.0,>=2.19.1 in /usr/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: python-slugify<5.0.0 in /home/vagrant/.local/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: gitpython<4.0.0 in /home/vagrant/.local/lib/python3.6/site-packages (from transifex-client)
Requirement already satisfied: chardet<5,>=3.0.2 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Requirement already satisfied: idna<3,>=2.5 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3.6/site-packages (from requests<3.0.0,>=2.19.1->transifex-client)
Requirement already satisfied: text-unidecode>=1.3 in /home/vagrant/.local/lib/python3.6/site-packages (from python-slugify<5.0.0->transifex-client)
Requirement already satisfied: typing-extensions>=3.7.4.0 in /usr/lib/python3.6/site-packages (from gitpython<4.0.0->transifex-client)
Requirement already satisfied: gitdb<5,>=4.0.1 in /home/vagrant/.local/lib/python3.6/site-packages (from gitpython<4.0.0->transifex-client)
Requirement already satisfied: smmap<5,>=3.0.1 in /home/vagrant/.local/lib/python3.6/site-packages (from gitdb<5,>=4.0.1->gitpython<4.0.0->transifex-client)
[vagrant@centos7-katello-devel-stable katello]$ make -C locale tx-update
make: Entering directory `/home/vagrant/katello/locale'
tx pull -f
tx INFO: Pulling translations for resource foreman.katello (source: locale/katello.pot)
tx INFO:  -> it: locale/it/katello.edit.po
tx INFO:  -> fr: locale/fr/katello.edit.po
tx INFO:  -> zh_CN: locale/zh_CN/katello.edit.po
tx INFO:  -> hi: locale/hi/katello.edit.po
tx INFO:  -> pa: locale/pa/katello.edit.po
tx INFO:  -> ru: locale/ru/katello.edit.po
tx INFO:  -> zh_TW: locale/zh_TW/katello.edit.po
tx INFO: Skipping 'cs' translation (file: locale/cs/katello.edit.po).
tx INFO:  -> mr: locale/mr/katello.edit.po
tx INFO:  -> ja: locale/ja/katello.edit.po
tx INFO:  -> or: locale/or/katello.edit.po
tx INFO:  -> ta: locale/ta/katello.edit.po
tx INFO:  -> pt_BR: locale/pt_BR/katello.edit.po
tx INFO:  -> ko: locale/ko/katello.edit.po
tx INFO:  -> bn: locale/bn/katello.edit.po
tx INFO:  -> pt: locale/pt/katello.edit.po
tx INFO:  -> kn: locale/kn/katello.edit.po
tx INFO:  -> es: locale/es/katello.edit.po
tx INFO:  -> gu: locale/gu/katello.edit.po
tx INFO:  -> te: locale/te/katello.edit.po
tx INFO:  -> de: locale/de/katello.edit.po
tx INFO: Done.
for f in ./bn/katello.edit.po ./cs/katello.edit.po ./de/katello.edit.po ./en/katello.edit.po ./es/katello.edit.po ./fr/katello.edit.po ./gu/katello.edit.po ./hi/katello.edit.po ./it/katello.edit.po ./ja/katello.edit.po ./kn/katello.edit.po ./ko/katello.edit.po ./mr/katello.edit.po ./or/katello.edit.po ./pa/katello.edit.po ./pt/katello.edit.po ./pt_BR/katello.edit.po ./ru/katello.edit.po ./ta/katello.edit.po ./te/katello.edit.po ./zh_CN/katello.edit.po ./zh_TW/katello.edit.po ; do \
	sed -i 's/^\("Project-Id-Version: \).*$/\1katello 2.4.0-RC1\\n"/' $f; \
done

Run rake plugin:gettext[katello] from the Foreman installation, then make -C locale mo-files to finish

make: Leaving directory `/home/vagrant/katello/locale'
```

This time it worked fine.

Reply:
Ah, I didn't even think about running it on EL7. I assumed it was on Fedora. Have you tried running `yum install python-slugify` instead?

Reply:
Where should I put this wiki?

Reply:
The way it was it was fine IMHO. Just drop the `install` line above it.

Reply:
i'm getting this on el7 and getting this same python-slugify error.  The 'python-slugify' rpm wasn't installed so i installed it manually and still get the error :/

Reply:
yeah this seems completely broken on el7, maybe we should at least suggest to run it on fedora or el8?